From 0ca35bbfe799a913068d9a268cb0243a8f3a2c4b Mon Sep 17 00:00:00 2001
From: cynthia <wu.cynthia@inventec.com>
Date: Sun, 25 Sep 2022 17:58:09 +0800
Subject: [PATCH] Modify 'Notify' function, allow new value types as input

---
 types.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/types.hpp b/types.hpp
index 1840edf..dc6b318 100644
--- a/types.hpp
+++ b/types.hpp
@@ -30,7 +30,8 @@ class Manager;
 
 /** @brief Inventory manager supported property types. */
 using InterfaceVariantType =
-    std::variant<bool, size_t, int64_t, std::string, std::vector<uint8_t>>;
+    std::variant<bool, size_t, int64_t, uint16_t, std::string, std::vector<uint8_t>, std::vector<uint16_t>,
+                 std::vector<std::string>>;
 
 template <typename T>
 using InterfaceType = std::map<std::string, T>;
-- 
2.33.0

