From 74202d3a8a543546ceb31fe1599850ab6644f04a Mon Sep 17 00:00:00 2001
From: Tommy Lin <lin.tommysc@inventec.com>
Date: Tue, 11 Jan 2022 05:09:22 +0000
Subject: [PATCH 01/27] Empty base dn error

---
 redfish-core/lib/account_service.hpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/redfish-core/lib/account_service.hpp b/redfish-core/lib/account_service.hpp
index a844b61a..759939a6 100644
--- a/redfish-core/lib/account_service.hpp
+++ b/redfish-core/lib/account_service.hpp
@@ -1065,6 +1065,17 @@ inline void handleLDAPPatch(nlohmann::json& input,
                                              "BaseDistinguishedNames");
             return;
         }
+        else
+        {
+            std::string baseDNfront = (*baseDNList).front();
+            if ( baseDNfront.empty())
+            {
+            messages::propertyValueNotInList(asyncResp->res, "[""]",
+                                                "BaseDistinguishedNames");
+            return;
+
+            }
+        }
     }
 
     // nothing to update, then return
-- 
2.34.1

