From 5aa952fa83ad337695b176771a02a3a058f73321 Mon Sep 17 00:00:00 2001
From: Tommy Lin <lin.tommysc@inventec.com>
Date: Tue, 11 Jan 2022 06:46:08 +0000
Subject: [PATCH] Boot flag

---
 redfish-core/lib/systems.hpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/redfish-core/lib/systems.hpp b/redfish-core/lib/systems.hpp
index 81403370..f7e54855 100644
--- a/redfish-core/lib/systems.hpp
+++ b/redfish-core/lib/systems.hpp
@@ -602,6 +602,11 @@ inline std::string dbusToRfBootSource(const std::string& dbusSource)
     {
         return "Cd";
     }
+    if (dbusSource ==
+        "xyz.openbmc_project.Control.Boot.Source.Sources.Diagnostic")
+    {
+        return "Diags";
+    }
     if (dbusSource == "xyz.openbmc_project.Control.Boot.Source.Sources.Network")
     {
         return "Pxe";
@@ -771,7 +776,7 @@ inline int assignBootParameters(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
     }
     else if (rfSource == "Diags")
     {
-        bootMode = "xyz.openbmc_project.Control.Boot.Mode.Modes.Safe";
+        bootSource = "xyz.openbmc_project.Control.Boot.Source.Sources.Diagnostic";
     }
     else if (rfSource == "Cd")
     {
-- 
2.33.0

