From 7ca5f0f210e3b82d8e228d79b6640826ff225912 Mon Sep 17 00:00:00 2001
From: Tommy Lin <Lin.TommySC@inventec.com>
Date: Tue, 11 Jul 2023 14:19:35 +0800
Subject: [PATCH] support smbios with pcimbox on nuvoton

---
 include/mdrv2.hpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/mdrv2.hpp b/include/mdrv2.hpp
index 86de51a..c34e0ed 100644
--- a/include/mdrv2.hpp
+++ b/include/mdrv2.hpp
@@ -59,9 +59,9 @@ static constexpr const char* systemInterface =
     "xyz.openbmc_project.Inventory.Item.System";
 constexpr const int limitEntryLen = 0xff;
 
-static constexpr const char* vgaMemPath = "/dev/vgasharedmem";
-static constexpr const uint32_t vgaMemPhyAddr = 0x0;
-static constexpr const uint32_t vgaMemPhySize = 64 * 1024;
+static constexpr const char* vgaMemPath = "/dev/mem";
+static constexpr const uint32_t vgaMemPhyAddr = 0xF0848000;
+static constexpr const uint32_t vgaMemPhySize = 0x3F00;
 
 class MDRV2 :
     sdbusplus::server::object_t<
-- 
2.25.1

