From 7e4831b17a7c073e2574e74e5b597de32e5ed107 Mon Sep 17 00:00:00 2001
From: Wein <kuo.wein@inventec.com>
Date: Fri, 13 Jan 2023 15:44:21 +0800
Subject: [PATCH] Custom modification of ACC100 for PLDMD

---
 src/numeric_sensor_handler.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/numeric_sensor_handler.cpp b/src/numeric_sensor_handler.cpp
index c63b2b2..49e44b0 100644
--- a/src/numeric_sensor_handler.cpp
+++ b/src/numeric_sensor_handler.cpp
@@ -108,7 +108,7 @@ bool NumericSensorHandler::setNumericSensorEnable(
     if (!validatePLDMRespDecode(_tid, rc, completionCode,
                                 "SetNumericSensorEnable"))
     {
-        return false;
+        //return false;
     }
 
     phosphor::logging::log<phosphor::logging::level::DEBUG>(
-- 
2.33.0

