From 2f73ce5e8b1ad7eded1b61cb8c13f86be5741c08 Mon Sep 17 00:00:00 2001
From: Wein <kuo.wein@inventec.com>
Date: Fri, 13 Jan 2023 16:25:00 +0800
Subject: [PATCH] Custom modification of BlueField2 for PLDMD

---
 src/firmware_update.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/firmware_update.cpp b/src/firmware_update.cpp
index ed39d36..b568956 100644
--- a/src/firmware_update.cpp
+++ b/src/firmware_update.cpp
@@ -2129,7 +2129,7 @@ int FWUpdate::runUpdate(const boost::asio::yield_context yield)
 
         expectedCmd = PLDM_VERIFY_COMPLETE;
 
-        startTimer(yield, fdCmdTimeout);
+        startTimer(yield, 50000);
 
         if (!fdReqMatched)
         {
@@ -2169,7 +2169,7 @@ int FWUpdate::runUpdate(const boost::asio::yield_context yield)
 
         expectedCmd = PLDM_APPLY_COMPLETE;
 
-        startTimer(yield, fdCmdTimeout);
+        startTimer(yield, 50000);
 
         if (!fdReqMatched)
         {
-- 
2.33.0

