From 096cf5799a53119e3d24cae10b7cc18ceefaa36f Mon Sep 17 00:00:00 2001
From: "Alang.Huang" <Huang.Alang@inventec.com>
Date: Sat, 6 Mar 2021 01:59:30 +0000
Subject: [PATCH 08/25] disable whitelist filter by commenting out
 registerFilter()

---
 src/whitelist-filter.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/whitelist-filter.cpp b/src/whitelist-filter.cpp
index 5b8eb27..9ac9289 100644
--- a/src/whitelist-filter.cpp
+++ b/src/whitelist-filter.cpp
@@ -107,13 +107,17 @@ WhitelistFilter::WhitelistFilter()
 {
     bus = getSdBus();
 
+    #if 0
     log<level::INFO>("Loading whitelist filter");
-
     ipmi::registerFilter(ipmi::prioOpenBmcBase,
                          [this](ipmi::message::Request::ptr request) {
                              return filterMessage(request);
                          });
+    #else
+
+    log<level::INFO>("Disable whitelist filter");
 
+    #endif
     channelSMM = getSMMChannel();
     // wait until io->run is going to fetch RestrictionMode
     post_work([this]() { postInit(); });
-- 
2.33.0

