From e7c14e64c9c698e8e3b16550f0684fc61214c9c2 Mon Sep 17 00:00:00 2001
From: Matt Cheng <cheng.matt@inventec.com>
Date: Mon, 24 Jul 2023 16:21:09 +0800
Subject: [PATCH] Add watchdog time

---
 drivers/watchdog/npcm_wdt.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/watchdog/npcm_wdt.c b/drivers/watchdog/npcm_wdt.c
index 622b30566a5b..5d12f7f04a08 100644
--- a/drivers/watchdog/npcm_wdt.c
+++ b/drivers/watchdog/npcm_wdt.c
@@ -142,8 +142,8 @@ static int npcm_wdt_set_timeout(struct watchdog_device *wdd,
 		wdd->timeout = 86;
 	else if (timeout < 173)
 		wdd->timeout = 172;
-	else if (timeout < 688)
-		wdd->timeout = 687;
+//	else if (timeout < 688)
+//		wdd->timeout = 687;
 	else
 		wdd->timeout = 2750;
 
-- 
2.25.1

