From 981505b31f5dbf07381b0825bc05b6d47f938846 Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Wed, 3 Aug 2022 16:57:15 +0800
Subject: [PATCH] driver: i2c: npcm7xx: don't check sda/scl status in driver
 probe

Signed-off-by: Joseph Liu <kwliu@nuvoton.com>
---
 drivers/i2c/busses/i2c-npcm7xx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/i2c/busses/i2c-npcm7xx.c b/drivers/i2c/busses/i2c-npcm7xx.c
index 435fe6652e66..1debc600d10e 100644
--- a/drivers/i2c/busses/i2c-npcm7xx.c
+++ b/drivers/i2c/busses/i2c-npcm7xx.c
@@ -1991,7 +1991,7 @@ static int npcm_i2c_init_module(struct npcm_i2c *bus, enum i2c_mode mode,
 	iowrite8(val, bus->reg + NPCM_I2CCTL1);
 
 	npcm_i2c_reset(bus);
-
+#if 0
 	/* check HW is OK: SDA and SCL should be high at this point. */
 	if ((npcm_i2c_get_SDA(&bus->adap) == 0) ||
 	    (npcm_i2c_get_SCL(&bus->adap) == 0)) {
@@ -2000,7 +2000,7 @@ static int npcm_i2c_init_module(struct npcm_i2c *bus, enum i2c_mode mode,
 			npcm_i2c_get_SCL(&bus->adap));
 		return -ENXIO;
 	}
-
+#endif
 	npcm_i2c_int_enable(bus, true);
 	return 0;
 }
-- 
2.34.1

