From 6e4f0fe9ff04350aeaa9ae4a12061508121490c0 Mon Sep 17 00:00:00 2001
From: "sun.cwsun" <sun.cwsun@inventec.com>
Date: Mon, 19 Sep 2022 16:17:17 +0800
Subject: [PATCH 30/31] user management framework concurrency access issue

---
 user_channel/user_mgmt.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/user_channel/user_mgmt.hpp b/user_channel/user_mgmt.hpp
index 602d549..5c34988 100644
--- a/user_channel/user_mgmt.hpp
+++ b/user_channel/user_mgmt.hpp
@@ -402,4 +402,4 @@ class UserAccess
     void cacheUserDataFile();
 };
 
-} // namespace ipmi
+} // namespace ipmi
\ No newline at end of file
-- 
2.33.0

